### PR TITLE
[Improvement](runtime-filter) build runtime filter before build hash table on join build probe

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
build runtime filter before build hash table on join build probe
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/exprs/runtime_filter_slots.h [hunk 1]
```diff
@@ -166,26 +166,24 @@ class VRuntimeFilterSlots {
...
                 continue;
             }
 
             int result_column_id = _build_expr_context[i]->get_last_result_column_id();
-            for (const auto* it : datas) {
-                auto column = it->get_by_position(result_column_id).column;
-                for (auto* filter : iter->second) {
-                    filter->insert_batch(column, 1);
-                }
+            auto column = block->get_by_position(result_column_id).column;
```
Comment (HappenLee) [inline be/src/exprs/runtime_filter_slots.h:177]: use reference ?

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 2]
```diff
@@ -949,9 +941,6 @@ void HashJoinNode::_set_build_ignore_flag(Block& block, const std::vector<int>&
 Status HashJoinNode::_process_build_block(RuntimeState* state, Block& block) {
     SCOPED_TIMER(_build_table_timer);
     size_t rows = block.rows();
-    if (UNLIKELY(rows == 0)) {
-        return Status::OK();
-    }
     COUNTER_UPDATE(_build_rows_counter, rows);
```
Comment (HappenLee) [inline be/src/vec/exec/join/vhash_join_node.cpp:944]: if (rows == 1) after `COUNTER_UPDATE`

File: be/src/pipeline/exec/hashjoin_build_sink.cpp [hunk 3]
```diff
@@ -482,29 +482,20 @@ Status HashJoinBuildSinkOperatorX::sink(RuntimeState* state, vectorized::Block*
...
-                          },
-                          [&](auto&& arg) -> Status {
-                              vectorized::ProcessRuntimeFilterBuild runtime_filter_build_process;
-                              return runtime_filter_build_process(state, arg, &local_state,
-                                                                  use_global_rf);
-                          }},
-                *local_state._shared_state->hash_table_variants);
+        auto ret = vectorized::process_runtime_filter_build(
+                state, local_state._shared_state->build_block.get(), &local_state, use_global_rf);
         if (!ret.ok()) {
```
Comment (HappenLee) [inline be/src/pipeline/exec/hashjoin_build_sink.cpp:490]: recheck the logic

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]

**Pull request merged**